### ISSUE TITLE
install_subdir() doesn't add DESTDIR to the installed-files list output

### ISSUE THREAD

**Issue by arteymix** [mesonbuild/meson#1070]
Here's the relevant packaging logs:

```
Installing subdir /home/guillaume/rpmbuild/BUILD/valum-0.3.0/x86_64-redhat-linux-gnu/docs/api to /usr/share/doc/valum-0.3.
Installing subdir /home/guillaume/rpmbuild/BUILD/valum-0.3.0/x86_64-redhat-linux-gnu/docs/devhelp/valum-0.3 to /usr/share/devhelp/books.
Installing subdir /home/guillaume/rpmbuild/BUILD/valum-0.3.0/x86_64-redhat-linux-gnu/docs/en to /usr/share/doc/valum-0.3/user.
Installing src/vsgi/servers/libvsgi-fastcgi.so to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/lib64/vsgi-0.3/servers/libvsgi-fastcgi.so
Installing src/vsgi/libvsgi-0.3.so to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/lib64/libvsgi-0.3.so
Installing src/vsgi/servers/libvsgi-http.so to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/lib64/vsgi-0.3/servers/libvsgi-http.so
Installing src/vsgi/servers/libvsgi-cgi.so to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/lib64/vsgi-0.3/servers/libvsgi-cgi.so
Installing src/vsgi/servers/libvsgi-scgi.so to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/lib64/vsgi-0.3/servers/libvsgi-scgi.so
Installing src/valum/libvalum-0.3.so to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/lib64/libvalum-0.3.so
Installing src/vsgi/servers/libvsgi-mock.so to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/lib64/vsgi-0.3/servers/libvsgi-mock.so
Installing vsgi.h to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/include/vsgi-0.3
Installing valum.h to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/include/valum-0.3
Installing /home/guillaume/rpmbuild/BUILD/valum-0.3.0/x86_64-redhat-linux-gnu/src/vsgi/vsgi-0.3.vapi to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/share/vala/vapi.
Installing /home/guillaume/rpmbuild/BUILD/valum-0.3.0/src/vsgi/vsgi-0.3.deps to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/share/vala/vapi.
Installing /home/guillaume/rpmbuild/BUILD/valum-0.3.0/x86_64-redhat-linux-gnu/meson-private/vsgi-0.3.pc to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/lib64/pkgconfig.
Installing /home/guillaume/rpmbuild/BUILD/valum-0.3.0/x86_64-redhat-linux-gnu/src/valum/valum-0.3.vapi to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/share/vala/vapi.
Installing /home/guillaume/rpmbuild/BUILD/valum-0.3.0/src/valum/valum-0.3.deps to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/share/vala/vapi.
Installing /home/guillaume/rpmbuild/BUILD/valum-0.3.0/x86_64-redhat-linux-gnu/meson-private/valum-0.3.pc to /home/guillaume/rpmbuild/BUILDROOT/valum-0.3-0.3.0-2.fc25.x86_64/usr/lib64/pkgconfig.
```

**Comment by ignatenkobrain** [comment]
that's not only related to meson, but DESTDIR is not enforced.

**Comment by arteymix** [comment]
However, DESTDIR is not set in https://github.com/mesonbuild/meson/blob/master/data/macros.meson. It works with absolute directories.

**Comment by ignatenkobrain** [comment]
http://pkgs.fedoraproject.org/cgit/rpms/ninja-build.git/tree/macros.ninja#n7

**Comment by arteymix** [comment]
You're right, I didn't notice the expansion there.

**Comment by arteymix** [comment]
After testing, I realized that this is not affecting packaging, but still the path has to be displayed/user correctly.

**Comment by eli-schwartz** [comment]
Duplicate of #3006

That one came later, but it got closed via #3011
